api/rooms/views/main.py

```diff
@@ -0,0 +1,38 @@
+from django.views     import View
+from django.http      import JsonResponse
+from django.db.models import Q
+
+from ..models import Room
+
+class RoomSuggestionView(View):
+    def get(self, request):
+        try: 
+            keyword_sort = request.GET.get('keyword_sort','서울')
+            offset       = int(request.GET.get('offset', 0))
+            limit        = int(request.GET.get('limit', 1))
+            
+            rooms = Room.objects.select_related('location','room_type','trade_type')\
+                    .prefetch_related('roomimage_set')\
+                    .filter(Q(location__dong__icontains=keyword_sort) | Q(location__city__icontains=keyword_sort))\
+                    .order_by('?')[offset:offset+limit]
```

Comment:
`filter()`는 `prefetch_related()` 앞에 붙여주는게 좋습니다!
`select_related`는 `JOIN` query를 사용한다고 했었죠?
filter는 select_related query시 WHERE 절이 붙어서 같이 조회됩니다!

`prefetch_related`는 추가로 쿼리를 날리기 때문에 filter 뒤에 붙여주는 것이 좋습니다!

```suggestion
            rooms = Room.objects.select_related('location','room_type','trade_type')\
                    .filter(Q(location__dong__icontains=keyword_sort) | Q(location__city__icontains=keyword_sort))\
                    .prefetch_related('roomimage_set')\
                    .order_by('?')[offset:offset+limit]
```

Reply:
넵 이해했습니다!

Reply:
검색하는 기능이 아니라서 keyword_sort의 범위는 dong으로 하나만 있으면 될거 같아서 다시 수정했습니다 혹시 여러개 넣으면 이점이 있을까요? 프론트에서 시,동 아무거나 넣어도 다 검색된다...? 이런 걸까요...?

Reply:
아래에 답변 남겨 놓았습니다~